modules/node-shared/src/main/scala/org/tessellation/node/shared/domain/transaction/FeeTransactionValidator.scala

```diff
@@ -0,0 +1,71 @@
+package org.tessellation.node.shared.domain.transaction
+
+import cats.data.{NonEmptyList, ValidatedNec}
+import cats.effect.Async
+import cats.syntax.all._
+
+import org.tessellation.currency.dataApplication.FeeTransaction
+import org.tessellation.ext.cats.syntax.validated._
+import org.tessellation.node.shared.domain.transaction.FeeTransactionValidator.FeeTransactionValidationErrorOr
+import org.tessellation.schema.address.Address
+import org.tessellation.security.signature.SignedValidator.SignedValidationError
+import org.tessellation.security.signature.{Signed, SignedValidator}
+
+import derevo.cats.{eqv, show}
+import derevo.derive
+
+trait FeeTransactionValidator[F[_]] {
+  def validate(signedTransaction: Signed[FeeTransaction]): F[FeeTransactionValidationErrorOr[Signed[FeeTransaction]]]
+  def validateMultiple(
+    signedTransactions: NonEmptyList[Signed[FeeTransaction]]
+  ): F[FeeTransactionValidationErrorOr[NonEmptyList[Signed[FeeTransaction]]]]
+}
+
+object FeeTransactionValidator {
+  def make[F[_]: Async](
+    signedValidator: SignedValidator[F]
+  ): FeeTransactionValidator[F] =
+    new FeeTransactionValidator[F] {
+      def validate(
+        signedTransaction: Signed[FeeTransaction]
+      ): F[FeeTransactionValidationErrorOr[Signed[FeeTransaction]]] =
+        for {
+          srcAddressSignatureV <- validateSourceAddressSignature(signedTransaction)
+          differentSrcAndDstV = validateDifferentSourceAndDestinationAddress(signedTransaction)
+        } yield
+          srcAddressSignatureV
+            .productR(differentSrcAndDstV)
+
+      def validateMultiple(
+        signedTransactions: NonEmptyList[Signed[FeeTransaction]]
+      ): F[FeeTransactionValidationErrorOr[NonEmptyList[Signed[FeeTransaction]]]] =
+        signedTransactions
+          .traverse(validate)
+          .map { validatedList =>
```

Comment:
`.map(_.sequence)`

Reply:
btw - isn't map.sequence a traverse? maybe use `.flatTraverse(validate)`?

Reply:
<img width="881" alt="Screenshot 2024-10-01 at 08 56 40" src="https://github.com/user-attachments/assets/cbd481db-8d94-48c8-8ef7-30ec531d1f58">
It fails when I use flatTraverse